src/test/objectstore/store_test_fixture.cc

```diff
@@ -56,10 +57,18 @@ void StoreTestFixture::SetUp()
 #endif
   ASSERT_EQ(0, store->mkfs());
   ASSERT_EQ(0, store->mount());
+
+  // we keep this stuff 'unsafe' out of test case scope to be able to update ANY
+  // config settings. Hence setting it to 'safe' here to proceed with the test
+  // case
+  g_conf->set_safe_to_start_threads();
```

Comment:
@liewegas - the idea and rationale of this stuff is to provide a way to adjust config settings that are prohibited in run-time, primarily bluestore_min_alloc_size.
Instead of introducing some additional flags/options to md_config_t I decided to rely on this flag. The idea is to clear it out of test case scope - see main() and TeadDown and set in SetUp. Once cleared it allows to set desired config parameters - that we tend to do *prior* to  SetUp in relevant test cases, see StartDeferred() stuff. And setting it here forbids config updates from test case body  as it was before. Moreover that didn't and wouldn't work anyway. And new SetVal allows to detect such issues by using set_value_or_die.
So long story short - this way we're able to update that 'protected' parameters out of test case only.

Reply:
ok!